dlog/plonk/src/index.rs

```diff
@@ -131,7 +131,7 @@ where
         srs: SRSSpec<'a, G>,
     ) -> Self {
         let srs = SRSValue::create(srs);
-        let max_poly_size = srs.get_ref().g.len();
+        let max_poly_size = cs.domain.d1.size as usize;
```

Comment:
can you explain this change? My understanding is that this is used to know what is the threshold before we have to split our polynomials. How is that related with d1's size?

Reply:
added a comment explaining

Reply:
I'm still not sure I understand, you're assuming:

* `cs.domain.d1.size` is going to be smaller than the SRS
* `cs.domain.d1.size` is going to be a perfect log2

In that case, why aren't we creating an SRS of this size, or trusting that it was created with the correct size in mind? (and keep using `srs.get_ref().g.len()`)

Reply:
(also the assert is useless right below if you set `max_poly_size = cs.domain.d1.size`